src/nvim/tui/term_input.inl

```diff
@@ -210,13 +209,29 @@ static bool handle_forced_escape(TermInput *input)
 
 static void read_cb(RStream *rstream, void *rstream_data, bool eof)
 {
+  TermInput *input = rstream_data;
+
   if (eof) {
-    input_done();
+    if (input->in_fd == 0 && !os_isatty(0) && os_isatty(2)) {
+      // Started reading from stdin which is not a pty but failed. Switch to
+      // stderr since it is a pty.
```

Comment:
I think the difference between is not noticeable by the program connected to the terminal:

- tty: master side is connected a physical terminal or a kernel terminal emulatorl(the linux console which you can select with alt+f{1-6} for example)
- pty: master side is connected to another user space terminal emulator such a xterm, tmux or neovim.

For consistency, I will change the PR title